docs/server/source/server-reference/configuration.md

```diff
@@ -102,6 +110,18 @@ The settings with names of the form `database.*` are for the database backend
 * `database.replicaset` is only relevant if using MongoDB; it's the name of the MongoDB replica set, e.g. `bigchain-rs`.
 * `database.connection_timeout` is the maximum number of milliseconds that BigchainDB will wait before giving up on one attempt to connect to the database backend.
 * `database.max_tries` is the maximum number of times that BigchainDB will try to establish a connection with the database backend. If 0, then it will try forever.
+* `database.ssl` is a flag that determines if BigchainDB connects to the
+  backend database over SSL or not. This can be set to any value (the presence of a value
+  qualifies as set or true in this case) to enable SSL connections. Do not
+  specify this parameter to connect over plaintext connection.
+  Note: This parameter is only supported for the MongoDB backend currently.
+* `database.login` and `database.password` are the login and password used to
+  authenticate to the database before performing any operations, specified in
+  plaintext.
+* `database.ca_cert`, `database.certfile`, `database.keyfile` and `database.crlfile` are the paths to the CA, signed certificate, private key and certificate revocation list files respectively.
+  Note: These parameters are only supported for the MongoDB backend currently.
+* `database.keyfile_passphrase` is the private key decryption passphrase, specified in plaintext.
+  Note: This parameter is only supported for the MongoDB backend currently.
```

Comment:
I have added a note that some params are only supported for the MongoDB backend.
We can modify docs when we begin to support both MongoDB and RethinkDB.